### PR TITLE
Allow passing an api_client kwarg to BotBase

### PR DESCRIPTION
**Pull request by ChrisLovering**

### CODE REVIEW

File: botcore/_bot.py [hunk 1]
```diff
@@ -49,9 +50,10 @@ def __init__(
             guild_id: The ID of the guild use for :func:`wait_until_guild_available`.
             allowed_roles: A list of role IDs that the bot is allowed to mention.
             http_session (aiohttp.ClientSession): The session to use for the bot.
-            redis_session: The
-                ``[async_rediscache.RedisSession](https://github.com/SebastiaanZ/async-rediscache#creating-a-redissession)``
-                to use for the bot.
+            redis_session: The `async_rediscache.RedisSession`_ to use for the bot.
+            api_client: The api_client instance to use for the bot.
```
Comment (MarkKoz) [inline botcore/_bot.py:54]: Shouldn't it be something like this (not sure how exactly sphinx wants it)?
```suggestion
            api_client: The `APIClient` instance to use for the bot.
```

File: botcore/_bot.py [hunk 2]
```diff
@@ -49,9 +50,10 @@ def __init__(
             guild_id: The ID of the guild use for :func:`wait_until_guild_available`.
             allowed_roles: A list of role IDs that the bot is allowed to mention.
             http_session (aiohttp.ClientSession): The session to use for the bot.
-            redis_session: The
-                ``[async_rediscache.RedisSession](https://github.com/SebastiaanZ/async-rediscache#creating-a-redissession)``
-                to use for the bot.
+            redis_session: The `async_rediscache.RedisSession`_ to use for the bot.
```
Comment (MarkKoz) [inline botcore/_bot.py:53]: Is that trailing underscore supposed to be there?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 5.0.4 18th April 2022
+- Feature: Allow passing an `api_client` to `BotBase`.`__init__` to specify an api client instance to use.
```
Comment (MarkKoz) [inline CHANGELOG.md:4]:
```suggestion
- Feature: Allow passing an `api_client` to `BotBase`.`__init__` to specify an `APIClient` instance to use.
```

File: botcore/_bot.py (hunk 2 above)
Comment (ChrisLovering) [reply botcore/_bot.py:53]: Yea, it "names" the object, allowing for the hyper linking at the end of the docstring.

File: botcore/_bot.py (hunk 1 above)
Comment (ChrisLovering) [reply botcore/_bot.py:54]: Changed in forcepush

File: CHANGELOG.md (hunk 3 above)
Comment (ChrisLovering) [reply CHANGELOG.md:4]: Changed in forcepush

**Review by MarkKoz — APPROVED** [review]
I guess netlify deployments have just been broken for a while now?

**Comment by ChrisLovering** [comment]
> I guess netlify deployments have just been broken for a while now?

Yea seems like it, see https://canary.discord.com/channels/267624335836053506/411200599653351425/965903442159734784 and down for a discussion about it in core dev.

**Pull request merged**